go/geth_adapter/adapter_test.go

```diff
@@ -466,3 +472,201 @@ func TestRunContextAdapter_bigIntToWord(t *testing.T) {
+		},
+		"Paris": {
+			block: parisBlock,
+			time:  uint64(0),
+			want:  tosca.R11_Paris,
+		},
+		"Shanghai": {
+			block: parisBlock,
+			time:  shanghaiTime,
+			want:  tosca.R12_Shanghai,
+		},
+		"Cancun": {
+			block: parisBlock,
+			time:  cancunTime,
+			want:  tosca.R13_Cancun,
+		},
+	}
+
+	chainConfig := &params.ChainConfig{
+		ChainID:            big.NewInt(42),
+		IstanbulBlock:      istanbulBlock,
+		LondonBlock:        londonBlock,
+		BerlinBlock:        berlinBlock,
+		MergeNetsplitBlock: parisBlock,
+		ShanghaiTime:       &shanghaiTime,
+		CancunTime:         &cancunTime,
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			got := convertRevision(chainConfig, test.block, test.time)
+			if got != test.want {
+				t.Errorf("Conversion returned wrong value, expected %v, got %v", test.want, got)
+			}
+		})
+	}
+}
+
+func TestRunContextAdapter_gethToVMErrors(t *testing.T) {
+	gas := tosca.Gas(42)
+	otherError := fmt.Errorf("other error")
+	tests := map[string]struct {
+		input      error
+		wantResult tosca.CallResult
+		wantError  error
+	}{
+		"nil": {
+			input: nil,
+		},
+		"insufficientBalance": {
+			input:      geth.ErrInsufficientBalance,
+			wantResult: tosca.CallResult{GasLeft: gas},
+			wantError:  nil,
+		},
+		"OutOfGas": {
+			input:      geth.ErrOutOfGas,
+			wantResult: tosca.CallResult{},
+			wantError:  nil,
+		},
+		"stackUnderflow": {
+			input:      &geth.ErrStackUnderflow{},
+			wantResult: tosca.CallResult{},
+			wantError:  nil,
+		},
+		"stackOverflow": {
+			input:      &geth.ErrStackOverflow{},
+			wantResult: tosca.CallResult{},
+			wantError:  nil,
+		},
+		"invalidOpCode": {
+			input:      &geth.ErrInvalidOpCode{},
+			wantResult: tosca.CallResult{},
+			wantError:  nil,
+		},
+		"other": {
+			input:      otherError,
+			wantResult: tosca.CallResult{},
+			wantError:  otherError,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			gotResult, gotErr := gethToVMErrors(test.input, gas)
+			if !errors.Is(gotErr, test.wantError) {
+				t.Errorf("Unexpected error: expected %v, got %v", test.wantError, gotErr)
+			}
+			reflect.DeepEqual(gotResult, test.wantResult)
+		})
+	}
+}
+
+func TestAdapter_ReadOnlyIsSetAndResetCorrectly(t *testing.T) {
+	tests := map[string]bool{
+		"readOnly":    true,
+		"notReadOnly": false,
+	}
+	recipient := tosca.Address{0x42}
+	depth := 42
+	gas := uint64(42)
+	for name, readOnly := range tests {
+		t.Run(name, func(t *testing.T) {
+			setGas := encodeReadOnlyInGas(gas, recipient, readOnly)
+			gotReadOnly, unsetGas := decodeReadOnlyFromGas(depth, readOnly, setGas)
+
+			if unsetGas != gas {
+				t.Errorf("Gas was not set or unset correctly, expected %v, got %v", gas, unsetGas)
+			}
+			if gotReadOnly != readOnly {
+				t.Errorf("ReadOnly was not set or unset correctly, expected %v, got %v", readOnly, gotReadOnly)
+			}
+		})
+	}
+}
+
+func TestGethInterpreterAdapter_RefundShiftIsRevertedCorrectly(t *testing.T) {
```

Comment:
nit: `TestGethInterpreterAdapter_RefundShiftIsReverted` 
I am having a crusade against unnecessary adjectives, our code should be correct ;)